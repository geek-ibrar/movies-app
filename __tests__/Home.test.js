import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { useDispatch, useSelector } from 'react-redux';
import Home from '../app/(tabs)/index';

const initialState = {
  dataEnded: false,
  loadingMovies: false,
  moviesList: [],
  favouritesList: [],
  pageNo: 1
}

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe('Home', () => {
  const mockDispatch = jest.fn();

  beforeEach(() => {
    // Reset mocks before each test
    useDispatch.mockReturnValue(mockDispatch);
    useSelector.mockImplementation((selector) =>
      selector({
        movies: initialState, // Mocking the initial state of the counter
      })
    );
  });

  afterEach(() => {
    jest.clearAllMocks(); // Clean up after each test
  });

  it('renders correctly', () => {
    const tree = render(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
