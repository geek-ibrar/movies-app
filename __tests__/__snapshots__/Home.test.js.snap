// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "padding": 20,
          "paddingBottom": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgb(238, 232, 244)",
            "borderRadius": 28,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="search-bar-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgb(238, 232, 244)",
              "borderRadius": 28,
              "flex": undefined,
              "flexDirection": "row",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="search-bar-container"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderRadius": 20,
                "height": 40,
                "margin": 6,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "width": 40,
              }
            }
            testID="search-bar-icon-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderColor": "rgba(121, 116, 126, 1)",
                  "borderRadius": 20,
                  "borderWidth": 0,
                  "elevation": 0,
                  "flex": 1,
                  "overflow": "hidden",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="search-bar-icon-container"
            >
              <View
                accessibilityComponentType="button"
                accessibilityLabel="search"
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityTraits="button"
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "alignItems": "center",
                        "flexGrow": 1,
                        "justifyContent": "center",
                      },
                      {
                        "borderRadius": 20,
                      },
                    ],
                  ]
                }
                testID="search-bar-icon"
              >
                <Text />
              </View>
            </View>
          </View>
          <TextInput
            accessibilityRole="search"
            keyboardAppearance="light"
            onChangeText={[Function]}
            placeholder="Movie Name"
            placeholderTextColor="rgba(28, 27, 31, 1)"
            returnKeyType="search"
            selectionColor="rgba(103, 80, 164, 1)"
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "fontSize": 18,
                  "minWidth": 0,
                  "paddingLeft": 8,
                  "textAlign": "left",
                },
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                  "lineHeight": 0,
                },
                {
                  "minHeight": 56,
                  "paddingLeft": 0,
                },
                undefined,
              ]
            }
            testID="search-bar"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            pointerEvents="none"
            style={
              [
                {
                  "marginLeft": 16,
                  "position": "absolute",
                  "right": 0,
                },
                false,
              ]
            }
            testID="search-bar-icon-wrapper"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 6,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "width": 40,
                }
              }
              testID="search-bar-clear-icon-container-outer-layer"
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderColor": "rgba(121, 116, 126, 1)",
                    "borderRadius": 20,
                    "borderWidth": 0,
                    "elevation": 0,
                    "flex": 1,
                    "overflow": "hidden",
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
                testID="search-bar-clear-icon-container"
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel="clear"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityTraits="button"
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  centered={true}
                  collapsable={false}
                  focusable={true}
                  hitSlop={
                    {
                      "bottom": 6,
                      "left": 6,
                      "right": 6,
                      "top": 6,
                    }
                  }
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      [
                        {
                          "alignItems": "center",
                          "flexGrow": 1,
                          "justifyContent": "center",
                        },
                        {
                          "borderRadius": 20,
                        },
                      ],
                    ]
                  }
                  testID="search-bar-clear-icon"
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListEmptyComponent={[Function]}
      ListFooterComponent={[Function]}
      contentContainerStyle={
        {
          "padding": 20,
          "paddingBottom": 0,
        }
      }
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.5}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <RCTRefreshControl />
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "margin": 30,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            No data
          </Text>
        </View>
        <View
          onLayout={[Function]}
        />
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
